### ISSUE TITLE
[Owned Entity] Foreign keys are being dropped and added on every new migration

### ISSUE THREAD

**Issue by sotasan** [dotnet/efcore#34600]
Hey,

every time I try to add a migration with `Add-Migration` or `dotnet ef migrations add`, it creates a non-empty migration with the same up and down every time. It tries to drop a foreign key and add it again shortly after. This behavior only happens on with the owned type. It happens using the Fluent API as well.

## Entities

```csharp
public class Test
{
    public int Id { get; set; }
    public IList<TestChild> Children { get; set; } = new List<TestChild>();
}

[Owned]
public class TestChild
{
    public int Id { get; set; }
}
```

## Migration

```csharp
protected override void Up(MigrationBuilder migrationBuilder)
{
    migrationBuilder.DropForeignKey(
        name: "FK_TestChild_Tests_TestId",
        table: "TestChild");

    migrationBuilder.AddForeignKey(
        name: "FK_TestChild_Tests_TestId",
        table: "TestChild",
        column: "TestId",
        principalTable: "Tests",
        principalColumn: "Id");
}

protected override void Down(MigrationBuilder migrationBuilder)
{
    migrationBuilder.DropForeignKey(
        name: "FK_TestChild_Tests_TestId",
        table: "TestChild");

    migrationBuilder.AddForeignKey(
        name: "FK_TestChild_Tests_TestId",
        table: "TestChild",
        column: "TestId",
        principalTable: "Tests",
        principalColumn: "Id",
        onDelete: ReferentialAction.Cascade);
}
```

## Verbose output

```
...
The index {'TestId'} was not created on entity type 'TestChild' as the properties are already covered by the index {'TestId', 'Id'}.
...
The property 'TestChild.TestId' was created in shadow state because there are no eligible CLR members with a matching name.
...
```

### Provider and version information

EF Core version: 7.0.15 (also tried 7.0.20)
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET 6.0
Operating system: Windows 11
IDE: Visual Studio 2022 17.11

**Comment by hayrikosem** [comment]
I think this is caused by onDeleteAction of the foreign key. Did you realize they are different on Up and Down methods. This happenned to me before and I figured it was caused by foreignkey configuration. I was configuring all the foreign keys onDeleteaction using reflection during model creation. This caused ownedentity foreignkeys' onDeleteAction to change. And it started creating this kind of migrations. 
I don't know it helps but when I changed owned entity foreignkey restriction it stoped creating this kind of migrations.
